public/script.js

```diff
@@ -1799,8 +1800,17 @@ export async function getCharacters() {
 
             characters[i]['chat'] = String(characters[i]['chat']);
         }
-        if (this_chid !== undefined) {
-            $('#avatar_url_pole').val(characters[this_chid].avatar);
+
+        if (previousAvatar) {
+            const newCharacterId = characters.findIndex(x => x.avatar === previousAvatar);
+            if (newCharacterId >= 0) {
+                setCharacterId(newCharacterId);
+                await unshallowCharacter(String(newCharacterId));
```

Comment:
There are other things that depend on chid. Found those on first glance:
https://github.com/SillyTavern/SillyTavern/blob/staging/public/script.js#L7851-L7854

You should likely just call `selectCharacterById`, which should handle all that. If you set the chid before, it runs into the second branch that is designed to "select the already selected character".
Yes, it will load the char into the char panel, if you just had the list open. That is unfortunate. But a trade-off I'd say is okay, so we do not have to rewrite every single piece where chid might be relevant.